packages/react-components/react-dialog/src/utils/useDisableBodyScroll.ts

```diff
@@ -1,55 +1,52 @@
 import { useFluent_unstable } from '@fluentui/react-shared-contexts';
-import { useCallback } from 'react';
+import { makeResetStyles } from '@griffel/react';
+import { useCallback, useRef } from 'react';
 
-const disableScrollElementProp = '__fluentDisableScrollElement' as const;
-
-type FluentDisableScrollElement = HTMLElement & {
-  [disableScrollElementProp]: {
-    count: number;
-    previousOverflowStyle: string;
-    previousPaddingRightStyle: string;
-  };
-};
+// this style must be applied to the html element to disable scrolling
+const useHTMLNoScrollStyles = makeResetStyles({
+  overflowY: ['hidden', 'clip'],
+});
 
 /**
- * hook that disables body scrolling through `overflow: hidden` CSS property
+ * hook that disables body scrolling through `overflowY: hidden` CSS property
  */
-export function useDisableBodyScroll() {
+export function useDisableBodyScroll(): {
+  disableBodyScroll: () => void;
+  enableBodyScroll: () => void;
+} {
+  const htmlNoScrollStyle = useHTMLNoScrollStyles();
   const { targetDocument } = useFluent_unstable();
-  return useCallback(() => {
-    if (targetDocument) {
-      return disableScroll(targetDocument.body);
+  const originalPaddingRight = useRef<string>('');
+
+  const disableBodyScroll = useCallback(() => {
+    if (!targetDocument) {
+      return;
     }
-  }, [targetDocument]);
-}
 
-/**
- * disables scrolling from a given element through `overflow: hidden` CSS property
- * @param target - element to disable scrolling from
- * @returns a method for enabling scrolling again
- */
-export function disableScroll(target: HTMLElement) {
-  const { clientWidth } = target.ownerDocument.documentElement;
-  const innerWidth = target.ownerDocument.defaultView?.innerWidth ?? 0;
-  assertIsDisableScrollElement(target);
-  if (target[disableScrollElementProp].count === 0) {
-    target.style.overflow = 'hidden';
-    target.style.paddingRight = `${innerWidth - clientWidth}px`;
-  }
-  target[disableScrollElementProp].count++;
-  return () => {
-    target[disableScrollElementProp].count--;
-    if (target[disableScrollElementProp].count === 0) {
-      target.style.overflow = target[disableScrollElementProp].previousOverflowStyle;
-      target.style.paddingRight = target[disableScrollElementProp].previousPaddingRightStyle;
+    targetDocument.documentElement.classList.add(htmlNoScrollStyle);
+
+    const window = targetDocument.defaultView;
+    const { clientWidth } = targetDocument.documentElement;
+    const innerWidth = window?.innerWidth ?? 0;
+
+    const verticalScrollbarGutter = `${innerWidth - clientWidth}px`;
+    originalPaddingRight.current = getComputedStyle(targetDocument.body).paddingRight;
```

Comment:
So here's the problem with this strategy, you need to use `getComputedStyle` which forces style recalc (https://gist.github.com/paulirish/5d52fb081b3570c81e3a#calling-getcomputedstyle), this is not super critical as it's not invoked multiple times but only once the dialog is opened.

Maybe we could add here some sort of verification if the browser supports `scrollbar-gutter` and if so we can avoid this approach entirely.

Reply:
I'm still against using scrollbar-gutter as the default since it forces the unsightly blank gutter artifact on the screen instead of just offsetting it with padding. But if y'all are sold on it, could we add a new property to Dialog called "hideScrollbarStrategy" with options like "scrollbarGutter" | "getComputedStyle" | "{staticPaddingString}". And then we can discuss what we want the default to be, but then app owners can override with their own personal choice and do stuff like 

`<Dialog hideScrollbarStrategy="scrollbarGutter" >`
`<Dialog hideScrollbarStrategy="getComputedStyle" >`
`<Dialog hideScrollbarStrategy="16px" >`

Because if scrollbarGutter is implemented as the only method for Dialog going forward, the first thing our apps would do is throw a useEffect on our Dialog implementations that hacky overrides it because we don't like the blank gutter artifact it adds to the screen

Reply:
Pushed up a change that shifts the disable/enable logic up to useDialog and adds the scrollbarOffset strategy prop to Dialog

Reply:
This is a feature request in contrast to a bug fix!

I'm against this approach of having a property on how to style things, style overrides exist to avoid this! Please let's roll this back to the way you implemented using `padding-right` instead of `scrollbar-gutter` , I do not wish to extend this bugfix into an API surface change, we can always create a feature request for that later.

Reply:
Let's revert this PR back to https://github.com/microsoft/fluentui/pull/31180/commits/6c7d7892700d75346722d5aad05bf04e814c443d, we can discuss `padding-right` vs `scrollbar-gutter` from that point later on

Reply:
Reverted :)